presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java

```diff
@@ -968,9 +975,9 @@ static OrcSelectiveRecordReader createCustomOrcSelectiveRecordReader(
                 columnTypes,
                 IntStream.range(0, types.size()).boxed().collect(toList()),
                 filters,
-                ImmutableList.of(),
-                ImmutableMap.of(),
-                ImmutableMap.of(),
+                filterFunctions,
```

Comment:
Why are you doing this? Is this because we are implementing  flatMap filters as filterfunctions instead?

Reply:
Presto can't write flat-map files, hence, we can't use `testRoundTrip` tests and can't write tests to run queries over flat-map data. Still, I wanted to get coverage for `getBlockView` and subfield pruning. To achieve that I added support for filter functions and subfield pruning to OrcTester.